### ISSUE TITLE
Seems functions encodeAddress and mnemonicToSecretKey generate different addresses.

### ISSUE THREAD

**Issue by codemaster101** [algorand/js-algorand-sdk#276]
So I was trying out the package with the following private and public key pairs (in hexadecimal form):
```
{ privateKey:
   'ca4ff9495d8db298403047fce0607860e26358d52bccaecd47ed484d31e1fdeb',
  publicKey:
   '03da10855b349dc30937a9b5451f7e1db6a3bd5b2312428a170758be49386a82ca' }
```

When I test the generation of an account and the retrieval of an address by this code:
(I provide the keys in their buffered formats)
```
const passphrase = algosdk.secretKeyToMnemonic(privateKey);
const account = algosdk.mnemonicToSecretKey(passphrase);
console.log(account.addr);
```
The address generated is: `ZANXZDI6Q7IZTYLNGPZQ423IYS5JBO4UBC2Z3ALGEJQXF6SLRNX7LV6F6Q`.

But when I try to generate the address from the public key I get:
```
const array = new Uint8Array(publicKey);
const address = algosdk.encodeAddress(array); // const address = algosdk.encodeAddress(publicKey) -> (same response)
console.log(address)
```
I get the the following response: `APNBBBK3GSO4GCJXVG2UKH36DW3KHPK3EMJEFCQXA5ML4SJYNKBMUR7KV6`.

Can someone tell me if I am using these in the correct way? If not, how to generate an address from a public key?
Am I using the generation of an account from a private key in the correct way?
https://github.com/algorand/js-algorand-sdk/blob/develop/src/account.js#L10
It seems encodeAddress does expect a public key.

**Comment by jasonpaulos** [comment]
@codemaster101 how are you generating the key pair? The private key you posted is only 32 bytes, but Algorand private keys should be 64 bytes. Since you started with an invalid private key, the operations you did on the private key are going to be invalid.

If you're doing key generation in the SDK, you can use `algosdk.generateAccount()` to generate a key pair.

**Comment by codemaster101** [comment]
@jasonpaulos RIGHT!
The key was secp256k1 and we needed ED25519 if I am not wrong (completely forgot about this). 

Maybe the SDK should reject the malformed key?
Because I was able to generate an account from it, send transactions from the secret key (I obviously did not match the privateKey and the secretKey to see if they were the same). 

Unfortunately, we don't use `generateAccount`, or similar functions in other blockchain SDKs ever. In fact I have not seen anyone use it, unless they are looking to test. Reason being, most of the companies I have worked with, want to reconstruct their wallets from a master seed ranging across a lot of chains. So it might be a good idea, to include that functionality?

I made it work like so (at first I was not using sodium to convert to a ED25519 pair):
```
const hdKey = HDKey.fromMasterSeed(bufferSeed);
const { privateKey, publicKey } = hdKey.derive(derivedPath);

async function x() {
	await _sodium.ready;
	const sodium = _sodium;
	const keypair = sodium.crypto_sign_seed_keypair(privateKey);
	return { privateKey: keypair.privateKey, publicKey: keypair.publicKey };
}

module.exports.x = x;
```
